orca-core/src/main/java/com/netflix/spinnaker/orca/pipeline/model/support/TriggerDeserializer.kt

```diff
@@ -163,6 +164,19 @@ class TriggerDeserializer :
           get("binaryUrl").textValue(),
           get("preferred").booleanValue()
         )
+        looksLikePipelineRefTrigger() -> PipelineRefTrigger(
+          get("type").textValue(),
+          get("correlationId")?.textValue(),
+          get("user")?.textValue() ?: "[anonymous]",
+          get("parameters")?.mapValue(parser) ?: mutableMapOf(),
+          get("artifacts")?.listValue(parser) ?: mutableListOf(),
+          get("notifications")?.listValue(parser) ?: mutableListOf(),
+          get("rebake")?.booleanValue() == true,
+          get("dryRun")?.booleanValue() == true,
+          get("strategy")?.booleanValue() == true,
+          get("parentExecutionId").textValue(),
+          get("parentPipelineStageId").textValue()
+        )
```

Comment:
Oh man, now I see what's wrong with my original PR...

https://github.com/spinnaker/orca/commit/ab5c7393245d3c5c5e9d514eac76c0e4cb1fe1c3#diff-6c52c34ee972ff5b7d41360cb22819831f9fc9b91063769a3665abf90bba1cdfR27

This custom serializer needs to statically register itself. One (kinda janky) way of doing that right now would be to do the following here: https://github.com/spinnaker/orca/commit/ab5c7393245d3c5c5e9d514eac76c0e4cb1fe1c3#diff-5948e4b7de0e8ea7c4c56962a84150934e979fbc96c7eb8a2e79b941fd45f2f5R143

```kotlin
@Bean
  fun pipelineRefTriggerDeserializer(): CustomTriggerDeserializerSupplier =
    PipelineRefTriggerDeserializerSupplier().also {
      TriggerDeserializer.customTriggerSuppliers.add(it)
    }
```

The proper correct way to do this would be to create an extension point, though... but this would suffice for the present.

Reply:
As for the missing data during serialization of this trigger... we need a way to get all of the unknown properties, without actually knowing them. We could do something to the effect of...

```patch
diff --git a/orca-sql/src/main/kotlin/com/netflix/spinnaker/orca/sql/PipelineRefTriggerDeserializerSupplier.kt b/orca-sql/src/main/kotlin/com/netflix/spinnaker/orca/sql/PipelineRefTriggerDeserializerSupplier.kt
index cd001d807..2c9a291bf 100644
--- a/orca-sql/src/main/kotlin/com/netflix/spinnaker/orca/sql/PipelineRefTriggerDeserializerSupplier.kt
+++ b/orca-sql/src/main/kotlin/com/netflix/spinnaker/orca/sql/PipelineRefTriggerDeserializerSupplier.kt
@@ -47,7 +47,11 @@ class PipelineRefTriggerDeserializerSupplier : CustomTriggerDeserializerSupplier
               isStrategy = get("strategy")?.booleanValue() == true,
               parentExecutionId = get("parentExecutionId").textValue(),
               parentPipelineStageId = get("parentPipelineStageId")?.textValue()
-            )
+            ).also {
+              fields().asSequence().filter { it.value != null }.forEach { entry ->
+                it.setOther(entry.key, entry.value)
+              }
+            }
           }
         }
         else -> throw SystemException("Trigger does not look like a pipelineRef trigger")
```

All of the other triggers know how to handle `other`.